### PR TITLE
Shortened Timespans (in commands)

### PR DESCRIPTION
**Pull request by Asleeepp**
### Description

Adds a Utility method to the Timespan class, where a timespan gets recognised in short form (1d, 1m, etc), and only works in the form of a command argument.

---
**Target Minecraft Versions:** any 
**Requirements:** none 
**Related Issues:** https://github.com/SkriptLang/Skript/issues/6327

### CODE REVIEW

File: src/main/java/ch/njol/skript/util/Timespan.java [hunk 1]
```diff
@@ -179,7 +196,50 @@ else if (length == 3 && !hasMs || length == 4) // HH:MM:SS[.ms]
...
+	 *
+	 * @param input   The input string representing the shortened timespan.
+	 * @param context The parse context to determine if parsing is allowed.
+	 * @return A Timespan object representing the parsed timespan, or null if parsing fails or not allowed in context.
+	 */
+	@Nullable
+	public static Timespan parseShortenedTimespan(String input, ParseContext context) {
+		if (context != ParseContext.COMMAND) {
+			return null;
+		}
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/util/Timespan.java:213]: No need for this context check. If a user is calling this function, they are doing so on purpose. Simply asking for a key that's never actually used seems silly.

File: src/main/java/ch/njol/skript/util/Timespan.java [hunk 2]
```diff
@@ -179,7 +196,50 @@ else if (length == 3 && !hasMs || length == 4) // HH:MM:SS[.ms]
...
+
+		long totalMillis = 0;
+		Matcher matcher = TIMESPAN_SHORTENED_PATTERN.matcher(input);
+
+		while (matcher.find()) {
+			int value = Integer.parseInt(matcher.group(1));
+			char unit = matcher.group(2).charAt(0);
+
+			switch (unit) {
+				case 'd':
```
Comment (sovdeeth) [inline src/main/java/ch/njol/skript/util/Timespan.java:223]: How does this handle `15doubles`?

**Review by sovdeeth — CHANGES_REQUESTED** [review]
Needs tests

File: src/main/java/ch/njol/skript/util/Timespan.java [hunk 3]
```diff
@@ -179,7 +196,50 @@ else if (length == 3 && !hasMs || length == 4) // HH:MM:SS[.ms]
...
+		if (context != ParseContext.COMMAND) {
+			return null;
+		}
+
+		long totalMillis = 0;
+		Matcher matcher = TIMESPAN_SHORTENED_PATTERN.matcher(input);
+
+		while (matcher.find()) {
+			int value = Integer.parseInt(matcher.group(1));
+			char unit = matcher.group(2).charAt(0);
```
Comment (Moderocky) [inline src/main/java/ch/njol/skript/util/Timespan.java:220]: The regex pattern will pick up a match within a larger string (e.g. `abc(1d)ef`) which is good if you want to extract multiple matches (like `1d 2h`) but will also pick up things it probably shouldn't (e.g. `(10m)onths`).
image
You can either limit this to just one match of the entire input (using the ^ and $ start and end tokens) or you can use some logic to make sure it's not the start of a larger text (e.g. `(?!\\w)`).

**Review by Moderocky — CHANGES_REQUESTED** [review]

File: src/main/java/ch/njol/skript/util/Timespan.java (hunk 3 above)
Comment (Asleeepp) [reply src/main/java/ch/njol/skript/util/Timespan.java:220]: Sovde told me to go a different route, and allow shorter versions in the default.lang, and then let the parser handle it. Which i'll try to fix up and finish today, as i didnt have much time yesterday.

**Pull request closed without merging**